src/material/datepicker/date-validators.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ValidatorFn, ValidationErrors, AbstractControl} from '@angular/forms';
+import {DateAdapter, MatDateFormats} from '@angular/material/core';
+
+/**
+ * @description
+ * Provides a set of date validators that can be used by form controls.
+ */
+// @dynamic
```

Comment:
I don't think this is a good idea, it might mask real errors. Instead of directly returning the `() => {}`'s below, just assign them to a variable and return it. (`const validator = ...; return validator;`). That should resolve the errors you were seeing.

Reply:
Sorry for the confusion, I just talked to someone on the framework team about this, and it sounds like `// @dynamic` is indeed the preferred way to handle this.

Reply:
hope I added correct docs. please feel free to request change format of tsdoc comment if find useful